src/components/run-button-container.js

```diff
@@ -450,21 +458,40 @@ export function RunButtonContainer({
     };
 
     const runnables = useMemo(() => {
-        let runnables = [
+        return [
             runnable[ComputingType.LOADFLOW],
-            runnable[ComputingType.SECURITY_ANALYSIS],
-            runnable[ComputingType.SENSITIVITY_ANALYSIS],
+            ...(!isUnavailableService(
+                unavailableOptionalServices,
+                OptionalServicesNames.SecurityAnalysis
+            )
+                ? [runnable[ComputingType.SECURITY_ANALYSIS]]
+                : []),
+            ...(!isUnavailableService(
+                unavailableOptionalServices,
+                OptionalServicesNames.SensitivityAnalysis
+            )
+                ? [runnable[ComputingType.SENSITIVITY_ANALYSIS]]
+                : []),
+            ...(!isUnavailableService(
+                unavailableOptionalServices,
+                OptionalServicesNames.ShortCircuit
+            ) && enableDeveloperMode
+                ? [runnable[ComputingType.SHORTCIRCUIT_ANALYSIS]]
+                : []),
+            ...(!isUnavailableService(
+                unavailableOptionalServices,
+                OptionalServicesNames.DynamicSimulation
+            ) && enableDeveloperMode
+                ? [runnable[ComputingType.DYNAMIC_SIMULATION]]
+                : []),
+            ...(!isUnavailableService(
+                unavailableOptionalServices,
+                OptionalServicesNames.VoltageInit
+            ) && enableDeveloperMode
+                ? [runnable[ComputingType.VOLTAGE_INIT]]
+                : []),
```

Comment:
I think it's not easy to read

make a function "shouldDisplayRunnable" that returns true/false depending on optionalService